apps/api/src/controllers/collaboratorsController.ts

```diff
@@ -113,33 +126,44 @@ export const createCollaborators = async ({
  * @param collaborator_id - ID of Collaborator to delete
  * @returns Success with Collaborator data record / Failure with Error.
  */
-export const deleteCollaborator = async ({ application_id, id }: { application_id: number; id: number }) => {
-	const database = getDbInstance();
-	const collaboratorsRepo: CollaboratorsService = collaboratorsSvc(database);
-	const applicationRepo: ApplicationService = applicationSvc(database);
-
-	const applicationResult = await applicationRepo.getApplicationById({ id: application_id });
-
-	if (!applicationResult.success) {
-		return applicationResult;
-	}
-
-	const application = applicationResult.data;
-
-	if (!(application.state === 'DRAFT')) {
-		return failure(`Can only add Collaborators when Application is in state DRAFT`, 'InvalidState');
-	}
-
-	const deleteResult = await collaboratorsRepo.deleteCollaborator({
-		id,
-	});
-
-	if (!deleteResult.success) {
-		return deleteResult;
+export const deleteCollaborator = async ({
+	application_id,
+	id,
+}: {
+	application_id: number;
+	id: number;
+}): AsyncResult<CollaboratorsResponseDTO[], 'INVALID_STATE_TRANSITION' | 'NOT_FOUND' | 'SYSTEM_ERROR'> => {
+	try {
+		const database = getDbInstance();
+		const collaboratorsRepo: CollaboratorsService = collaboratorsSvc(database);
+		const applicationRepo: ApplicationService = applicationSvc(database);
+
+		const applicationResult = await applicationRepo.getApplicationById({ id: application_id });
+
+		if (!applicationResult.success) {
+			return applicationResult;
+		}
+
+		const application = applicationResult.data;
+
+		// TODO: Valid states for actions should be handled through the appStateManager
+		if (!(application.state === 'DRAFT')) {
+			return failure('INVALID_STATE_TRANSITION', `Can only add Collaborators when Application is in state DRAFT`);
```

Comment:
Yeah there are a number of similar conditional checks in the app like this that could be moved into state manager functions. 
In this case though there's no state transition so do you think stateManager is necessary?

Reply:
this action is only valid when the application is in some states, so the action should be added to the state manager. doesn't need a state transition, just a check if the action can be pefformed